### DISCUSSION TITLE
What is the TypeScript definition of ISourceOptions with emitters?

### DISCUSSION THREAD

**Discussion by DigitalNaut** [matteobruni/tsparticles#5342] [Q&A]
Which type do I need to use for the options object to work with the emitters plugin in TypeScript?

```ts
const options: ISourceOptions = {}
```

`ISourceOptions` doesn't seem to include anything related to emitters and the `@tsparticles/plugin-emitters` package doesn't export an obvious type. Thanks!

Edit:

So I've found `IEmitterOptions` somewhere in a package and I'm currently using `ISourceOptions & IEmitterOptions` as the type definition. However, it's proving to be quite demanding with the types. Is this correct?

![image](https://github.com/tsparticles/tsparticles/assets/12903889/928dcd4f-6bcc-49d1-8b29-7d7c1290960d)

Is `ISourceOptions & RecursivePartial<IEmitterOptions>` more appropriate?

**Comment by TheDelta** [comment]
First of all thanks for the hint where the type is. ❤️  I was going crazy and the documentation is so horrible to find something.
`import { IEmitterOptions } from '@tsparticles/plugin-emitters/types/types';`

(migrating atm my angular project from tsparticles v2 (old namespace) to v3) and used this all the time. So to answer your questions: probably more appropriate :)
`ISourceOptions & RecursivePartial<IEmitterOptions>` (or in ngx context: `IParticlesProps & RecursivePartial<IEmitterOptions>` )
